src/plugins.ts

```diff
@@ -124,15 +133,24 @@ export default class Plugins {
     root: string,
     {prod = true}: { prod?: boolean } = {},
   ): Promise<void> {
-    if (fs.existsSync(path.join(root, 'yarn.lock'))) {
-      this.debug(`yarn.lock exists at ${root}. Installing prod dependencies`)
+    const doRefresh = async () => {
       // use yarn.lock to fetch dependencies
       await this.yarn.exec(prod ? ['--prod'] : [], {
         cwd: root,
         verbose: this.verbose,
       })
+    }
+
+    if (await fileExists(path.join(root, 'oclif.lock'))) {
+      this.debug(`oclif.lock exists at ${root}. Installing prod dependencies`)
```

Comment:
I think it would be worth adding a comment here explaining why we are doing this. If you came to plugin-plugins first it might be confusing why we are messing with an `oclif.lock` file